SceneRefAttributeValidator.cs

```diff
@@ -334,22 +332,52 @@ private static void ValidateRef(SceneRefAttribute attr, Component c, FieldInfo f
 
             if (IsEmptyOrNull(value, isArray))
             {
-                if (!attr.HasFlags(Flag.Optional))
+                if (attr.HasFlags(Flag.Optional)) return;
+                
+                var elementType = isArray ? fieldType.GetElementType() : fieldType;
+                elementType = typeof(ISerializableRef).IsAssignableFrom(elementType) ? elementType?.GetGenericArguments()[0] : elementType;
+                Debug.LogError($"{c.GetType().Name} missing required {elementType?.Name}{(isArray ? "[]" : "")} ref '{field.Name}'", c.gameObject);
+                return;
+            }
+
+            if (attr.HasFlags(Flag.Optional))
+            {
+                if (isArray)
                 {
-                    Type elementType = isArray ? fieldType.GetElementType() : fieldType;
-                    elementType = typeof(ISerializableRef).IsAssignableFrom(elementType) ? elementType?.GetGenericArguments()[0] : elementType;
-                    Debug.LogError($"{c.GetType().Name} missing required {elementType?.Name + (isArray ? "[]" : "")} ref '{field.Name}'", c.gameObject);
+                    Array a = (Array)value;
+                    for (int i = 0; i < a.Length; ++i)
+                    {
+                        object o = UnpackValue(a.GetValue(i));
+                        if (o == null)
+                            continue;
+
+                        if (o is MonoBehaviour m && m.gameObject == c.gameObject)
```

Comment:
Maybe I'm missing it but shouldn't this also check if the exclude self flag is set? (same below on line 365)